search-app/src/cmr/search/services/query_execution/facets/facets_v2_results_feature.clj

```diff
@@ -140,15 +140,20 @@
       (assoc (v2-facets-root concept-type) :has_children true :children v2-facets)
       (assoc (v2-facets-root concept-type) :has_children false))))
 
+(def facets-validation
+  "Mapping of concept type to the validator to run for that concept type."
+  {:granule granule-v2-facets/validator})
+
 (defmethod query-execution/pre-process-query-result-feature :facets-v2
-  [{:keys [query-string]} query _]
+  [{:keys [query-string] :as context} {:keys [concept-type facet-fields] :as query} _]
```

Comment:
This destructing in argument list is getting too complicated, move them inside the function as a `let` block.